App/Sagas/ImageSharingSagas.ts

```diff
@@ -145,6 +148,17 @@ export function * uploadPins (uuid: string) {
   }
 }
 
+export function * monitorForUploadsComplete(uuid: string) {
```

Comment:
This is cool. Once the uploads are started for a single share, this saga waits for them all to finish before calling `shareToThread`. Uses a special `waitFor` effect that I wrote.

Reply:
pro